lib/services/search_service.dart

```diff
@@ -0,0 +1,137 @@
+import 'package:dio/dio.dart';
+import 'package:logging/logging.dart';
+import 'package:photos/core/configuration.dart';
+import 'package:photos/core/event_bus.dart';
+import 'package:photos/core/network.dart';
+import 'package:photos/db/files_db.dart';
+import 'package:photos/events/local_photos_updated_event.dart';
+import 'package:photos/models/file.dart';
+import 'package:photos/models/location.dart';
+import 'package:photos/models/search/location_search_result.dart';
+import 'package:photos/services/user_service.dart';
+
+class SearchService {
+  List<File> _cachedFiles;
+  Future<List<File>> _future;
+  final _dio = Network.instance.getDio();
+  final _config = Configuration.instance;
+  final _logger = Logger((UserService).toString());
+
+  SearchService._privateConstructor();
+  static final SearchService instance = SearchService._privateConstructor();
+
+  Future<void> init() async {
+    // Intention of delay is to give more CPU cycles to other tasks
+    Future.delayed(const Duration(seconds: 5), () async {
+      // In case home screen loads before 5 seconds and user starts search, future will not be null
+      _future == null
+          ? FilesDB.instance.getAllFilesFromDB().then((value) {
+              _cachedFiles = value;
+            })
+          : null;
+    });
+
+    Bus.instance.on<LocalPhotosUpdatedEvent>().listen((event) {
+      _cachedFiles = null;
+      getAllFiles();
+    });
+  }
+
+  Future<List<File>> getAllFiles() async {
+    if (_cachedFiles != null) {
+      return _cachedFiles;
+    }
+    if (_future != null) {
+      return _future;
+    }
+    _future = _fetchAllFiles();
+    return _future;
+  }
+
+  Future<List<File>> _fetchAllFiles() async {
+    _cachedFiles = await FilesDB.instance.getAllFilesFromDB();
+    return _cachedFiles;
+  }
+
+  Future<List<File>> getFilesOnFilenameSearch(String query) async {
+    List<File> matchedFiles = [];
+    List<File> files = await getAllFiles();
+    //<20 to limit number of files in result
+    for (int i = 0; (i < files.length) && (matchedFiles.length < 20); i++) {
```

Comment:
Let's extract `20` to a constant with a meaningful name, so that the comment above can be avoided?